test/test_loader.py

```diff
@@ -0,0 +1,477 @@
+# coding=utf-8
+"""Test for the ``sopel.loader`` module."""
+import imp
+import inspect
+import os
+
+import pytest
+
+from sopel import loader, config, module
+
+
+MOCK_MODULE_CONTENT = """# coding=utf-8
+import sopel.module
+
+
+@sopel.module.commands("first")
+def first_command(bot, trigger):
+    pass
+
+
+@sopel.module.commands("second")
+def second_command(bot, trigger):
+    pass
+
+
+@sopel.module.interval(5)
+def interval5s(bot):
+    pass
+
+
+@sopel.module.interval(10)
+def interval10s(bot):
+    pass
+
+
+@sopel.module.url(r'.\\.example\\.com')
+def example_url(bot):
+    pass
+
+
+@sopel.module.event('TOPIC')
+def on_topic_command(bot):
+    pass
+
+
+def shutdown():
+    pass
+
+
+def ignored():
+    pass
+
+"""
+
+
+@pytest.fixture
+def func():
+    """Pytest fixture to get a function that will returns True all the time"""
```

Comment:
Since I don't expect to find any big things to change in this PR, I'll point out the little tiny ones like this grammatical mistake. :grin:

```suggestion
    """Pytest fixture to get a function that will return True all the time"""
```